engine/storage/src/org/apache/cloudstack/storage/datastore/ObjectInDataStoreManagerImpl.java

```diff
@@ -172,7 +175,11 @@ public DataObject create(DataObject obj, DataStore dataStore) {
                     if (snapshotDataStoreVO != null) {
                         ss.setParentSnapshotId(snapshotDataStoreVO.getSnapshotId());
                     }
-                    ss.setInstallPath(TemplateConstants.DEFAULT_SNAPSHOT_ROOT_DIR + "/" + snapshotDao.findById(obj.getId()).getAccountId() + "/" + snapshot.getVolumeId());
+                    String snapshotInstallPath = TemplateConstants.DEFAULT_SNAPSHOT_ROOT_DIR + "/" + snapshotDao.findById(obj.getId()).getAccountId() + "/" + snapshot.getVolumeId();
+                    if (snapshot.getHypervisorType().equals(HypervisorType.VMware)) {
```

Comment:
Is this really only true for vmware? It seems that adding a uuid to the path is a rather universal action in the ssvm, no?